protos/feast/core/SavedDataset.proto

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright 2021 The Feast Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+
+syntax = "proto3";
+
+package feast.core;
+option java_package = "feast.proto.core";
+option java_outer_classname = "SavedDatasetProto";
+option go_package = "github.com/feast-dev/feast/sdk/go/protos/feast/core";
+
+import "google/protobuf/timestamp.proto";
+import "feast/core/FeatureViewProjection.proto";
+import "feast/core/DataSource.proto";
+
+message SavedDatasetSpec {
+  // Name of the dataset. Must be unique since it's possible to overwrite dataset by name
+  string name = 1;
+
+  // Name of Feast project that this Dataset belongs to.
+  string project = 2;
+
+  // list of feature references with format "<view name>:<feature name>"
+  repeated string features = 3;
```

Comment:
1. FeatureViewProjection seems to be overkill here because all mappings have already happened during historical retrieval. So this dataset is already a projection. What we need to read it is just a list of columns.
2. Although it seems like a reasonable idea to replace the list of features with a feature service name there's one concern: feature service can be updated and the dataset is kinda immutable. So we'll need to infer the schema for dataset columns, instead of using this property and thus we can omit it completely.

Reply:
SGTM. Also, seems like we also might want to list the schema here too so it's easier to do type validation down the road? (and maybe even allow for string tags like feature views so users can add additional metadata for a dataset... like author of a dataset)

Reply:
Added tags to both SavedDataset proto & options